File: official/docs/csharp/current/pagination/get-next-page.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Newtonsoft.Json;
using EasyPost;
using EasyPost.Models.API;
using EasyPost.Parameters;

namespace EasyPostExamples
{
    public class Examples
    {
        public static async Task Main()
        {
            var client = new EasyPost.Client(new EasyPost.ClientConfiguration("EASYPOST_API_KEY"));

            // Get first page of results
            Parameters.Shipment.All parameters = new()
            {
                PageSize = 5
            };

            ShipmentCollection shipmentCollection = await client.Shipment.All(parameters);

            // Provide the previous results page to move onto the next page
            ShipmentCollection nextPage = await client.Shipment.GetNextPage(shipmentCollection)

            Console.WriteLine(JsonConvert.SerializeObject(nextPage, Formatting.Indented));
        }
    }
}

File: official/docs/golang/current/pagination/get-next-page.go
package example

import (
	"fmt"

	"github.com/EasyPost/easypost-go/v4"
)

func main() {
	client := easypost.New("EASYPOST_API_KEY")

	// Get first page of results
	shipments, _ := client.ListShipments(
		&easypost.ListShipmentsOptions{
			PageSize: 5,
		},
	)

	// Provide the previous results page to move onto the next page
	secondPage, _ := client.GetNextShipmentPage(shipments)

	// You can also ask for the next page to be of a specific size
	lastPage, _ := client.GetNextShipmentPageWithPageSize(secondPage, 10)

	fmt.Println(lastPage)
}
